owncloudApp/src/main/java/com/owncloud/android/MainApp.kt

```diff
@@ -207,8 +214,28 @@ class MainApp : Application() {
         )
     }
 
+    private fun handleRestrictions(activity: Activity) {
+        val restrictionsManager = activity.getSystemService(Context.RESTRICTIONS_SERVICE) as RestrictionsManager
+        val restrictions = restrictionsManager.applicationRestrictions
+        cacheRestrictions(activity, restrictions)
+    }
+
+    private fun cacheRestrictions(activity: Activity, restrictions: Bundle) {
+        val preferencesProvider = SharedPreferencesProviderImpl(activity)
+        if (restrictions.containsKey(CONFIGURATION_SERVER_URL)) {
+            val managedString = restrictions.getString(CONFIGURATION_SERVER_URL)
+            managedString?.let { preferencesProvider.putString(CONFIGURATION_SERVER_URL, it) }
+        }
+        if (restrictions.containsKey(CONFIGURATION_SERVER_URL_INPUT_VISIBILITY)) {
+            val managedBool = restrictions.getBoolean(CONFIGURATION_SERVER_URL_INPUT_VISIBILITY)
+            preferencesProvider.putBoolean(CONFIGURATION_SERVER_URL_INPUT_VISIBILITY, managedBool)
```

Comment:
A tiny question, Could `managedBool` be null? as `managedString` restriction? In this case, you should manage it.

Reply:
No, it can't hehe. In this case `getBoolean` assures that the value returned is not nullable, and that's why I check it in the previous line with `containsKey`.

Reply:
Perfect 👍🏻